File: Java/백준/Gold/2230. 수 고르기/수 고르기.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        int[] arr = new int[n];
        for (int i = 0; i<n; i++) {
            arr[i] = sc.nextInt();
        }

        Arrays.sort(arr);
 
        int pointer1 = 0;
        int pointer2 = 0;
        
        int answer = Integer.MAX_VALUE;
        while (pointer1 < n && pointer2 < n) {
            int diff = arr[pointer2] - arr[pointer1];

            if (diff >= m) {
                answer = Math.min(answer, diff);

                pointer1++;

                if (answer == m) break;
            } else {
                pointer2++;
            }
        }

        System.out.println(answer);
    }
}

File: Swift/백준/Gold/2230. 수 고르기/수 고르기.swift
import Foundation

let firstInput = readLine()!.split(separator: " ").map { Int($0)! }
let n: Int = firstInput[0]
let m: Int = firstInput[1]


var arr: [Int] = []
for i in 0..<n {
    if let line = readLine(), let num = Int(line) {
        arr.append(num)
    }
}

arr.sort()

var p1: Int = 0
var p2: Int = 0

var answer = Int.max

while p1 < n && p2 < n {
    let diff = arr[p2] - arr[p1]
    
    if diff >= m {
        answer = min(answer, diff)
        
        p1 += 1
        
        if answer == m {
            break
        }
    } else {
        p2 += 1
    }
}

print(answer)